dwave/samplers/__init__.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2018 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+"""
+A unified namespace for :class:`dimod.Sampler`s.
+
+This namespace is populated by entry points. To contribute a sampler, you
+must identify it in your setup.py file like so.
+
+.. code-block:: python
+
+    entry_points = {'dwave.samplers': ['YourSampler = import.path.to.sampler:YourSampler']}
+
+    setup(
+        entry_points=entry_points,
+        ...
+    )
+
+"""
+from pkg_resources import iter_entry_points
+
+# add the samplers to the module from entrypoints, name conflicts override
+globals().update((ep.name, ep.load()) for ep in iter_entry_points('dwave.samplers'))
```

Comment:
Oh, since now we have samplers in Hybrid as well, maybe we want to differentiate those from dimod samplers?

Reply:
Something like:
```
dwave.dimod.samplers
dwave.hybrid.samplers
```
Or think of a way to reconciliate the two. Same goes for `dwave.composites`.

Reply:
how about
```
dwave.samplers
dwave.composites
dwave.runnables  # this could be further subdivided
```

Reply:
I don't love it, but it could work.

On a somewhat related note, I'm getting fond of `from dwave import dimod, hybrid`.

Reply:
Are you proposing making dimod and hybrid into namespace packages? That would be easy to do, but at some point it would make sense to just make them all into a monolith package.

Or do you mean like the above where just the samplers/composites are present? I think it would be extremely confusing for `from dwave import hybrid` to contain different contents/structure than `import hybrid`.

Reply:
The former. I'm toying with the idea.